Jenkinsfile

```diff
@@ -1,17 +1,35 @@
 pipeline {
     agent none
+    triggers {
+        cron(env.BRANCH_NAME == 'master' | env.BRANCH_NAME == 'next' ? '@daily' : '')
+    }
+    options {
+        retry(1)
+        parallelsAlwaysFailFast()
+        timeout(time: 2, unit: 'HOURS')
+    }
     environment {
         UPDATE_CACHE = "true"
         DOCKER_CREDENTIALS = credentials('dockerhub')
         DOCKER_USERNAME = "${env.DOCKER_CREDENTIALS_USR}"
         DOCKER_PASSWORD = "${env.DOCKER_CREDENTIALS_PSW}"
         KONG_PACKAGE_NAME = "kong"
     }
-    triggers {
-        cron('@daily')
-    }
     stages {
         stage('Build Kong') {
+            when {
+                beforeAgent true
+                anyOf {
+                    allOf {
+                        buildingTag()
+                        not { triggeredBy 'TimerTrigger' }
+                    }
+                    allOf {
+                        triggeredBy 'TimerTrigger'
+                        anyOf { branch 'master'; branch 'next' }
+                    }
+                }
+            }
```

Comment:
this evaluates to

```
if (
  ( buildingtag && not cron ) ||
  ( ( branch = master || branch = next) && cron )
)

Reply:
Please put that in a comment on the file, too.

Reply:
added in f4485bbbaec09731108d6191dedaa6448702fdb6